test/pubsub/iam.js

```diff
@@ -0,0 +1,148 @@
+        assert.strictEqual(method, 'GET');
+
+        var expectedPath = RESOURCE + ':getIamPolicy';
+        assert.strictEqual(path, expectedPath);
+
+        assert.strictEqual(q, null);
+        assert.strictEqual(body, null);
+
+        done();
+      };
+
+      iam.getPolicy(assert.ifError);
+    });
+  });
+
+  describe('setPolicy', function() {
+    it('should throw an error if a policy is not supplied', function() {
+      assert.throws(function() {
+        iam.setPolicy(noop);
+      }, /A policy is required/);
+    });
+
+    it('should make the correct API request', function(done) {
+      var policy = { etag: 'ACAB' };
+
+      iam.makeReq_ = function(method, path, q, body) {
+        assert.strictEqual(method, 'POST');
+
+        var expectedPath = RESOURCE + ':setIamPolicy';
+        assert.strictEqual(path, expectedPath);
+
+        assert.strictEqual(q, null);
+
+        var expectedBody = { policy: policy };
+        assert.deepEqual(body, expectedBody);
+
+        done();
+      };
+
+      iam.setPolicy(policy, assert.ifError);
+    });
+  });
+
+  describe('testPermissions', function() {
+    it('should throw an error if permissions are missing', function() {
+      assert.throws(function() {
+        iam.testPermissions(noop);
+      }, /Permissions are required/);
+    });
+
+    it('should make the correct API request', function(done) {
+      var permissions = 'storage.bucket.list';
+
+      iam.makeReq_ = function(method, path, q, body) {
+        assert.strictEqual(method, 'POST');
+
+        var expectedPath = RESOURCE + ':testIamPermissions';
+        assert.strictEqual(path, expectedPath);
+
+        assert.strictEqual(q, null);
+
+        var expectedBody = { permissions: [permissions] };
+        assert.deepEqual(body, expectedBody);
+
+        done();
+      };
+
+      iam.testPermissions(permissions, assert.ifError);
+    });
+
+    it('should send an error back if the request fails', function(done) {
+      var permissions = ['storage.bucket.list'];
+      var error = new Error('Error.');
+      var fakeResponse = {};
+
+      iam.makeReq_ = function(method, path, q, body, callback) {
+        callback(error, fakeResponse);
+      };
+
+      iam.testPermissions(permissions, function(err, perms, resp) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(perms, null);
+        assert.strictEqual(resp, fakeResponse);
+        done();
+      });
+    });
+
+    it('should pass back an array if permissions is undefined', function(done) {
+      var permissions = ['storage.bucket.list'];
+      var fakeResponse = {};
+
+      iam.makeReq_ = function(method, path, q, body, callback) {
+        callback(null, fakeResponse);
+      };
+
+      iam.testPermissions(permissions, function(err, perms, resp) {
+        assert.ifError(err);
+        assert.deepEqual(perms, { 'storage.bucket.list': false });
+        assert.strictEqual(resp, fakeResponse);
+        done();
+      });
+    });
```

Comment:
Also test the success case: if the permission is true, check that the `perms` callback argument says it is true.